internal/events/kafka.go

```diff
@@ -84,13 +89,30 @@ func NewKafkaProducer(config KafkaConfig) (*KafkaProducer, error) {
 		return nil, fmt.Errorf("invalid kafka config: %w", err)
 	}
 
+	var tlsConfig *tls.Config
 	transport := kafka.DefaultTransport
+	if slices.Contains(SSLProtocols, config.SecurityProtocol) {
+		tlsConfig = &tls.Config{}
+		if config.SecurityProtocol == KafkaProtocolSSL {
```

Comment:
Good point! The `SSL` connection requires a Keypair (Access Key and Access Certificate) that the Kafka server will use to verify the client's identity (it means the authentication process).

The `SASL_SSL` uses the SSL to create an encrypted connection but the authentication process differs since you can use `PLAIN`, `GSSAPI (Kerberos)`.

Please refer to this: https://developer.confluent.io/courses/security/authentication-ssl-and-sasl-ssl/#:~:text=SSL%20is%20ideally%20suited%20for,different%20components%20across%20different%20environments.